File: 0035-search-insert-position/0035-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        if nums[-1]<target:
            return len(nums)
        elif target<=nums[0]:
            return 0
        else:            
            nums=nums+[target]
            nums.sort()
            start=0
            end=len(nums)-1
            res=0
            while start<=end:
                mid=(start+end)//2
                if nums[mid]==target:
                    res=mid
                    break
                elif nums[mid]<target:
                    start=mid+1
                else:
                    end=mid-1
            if nums[mid]==nums[mid-1]:
                return mid-1
            else:
                return mid
                

File: 0035-search-insert-position/0035-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int left=0;
        int right=nums.size();
        cout<<left<<right<<endl;
        if(nums[right-1]>=target){
        while(left<=right){
            int mid= left + (right-left)/2;
            if(nums[mid]==target){
                return mid;
            }
            else if(nums[mid]<target){
                left=mid+1;
            }
            else if(nums[mid]>target){
                right=mid-1;
            }
        }
        cout<<"outside";
        return right+1;
        }
        else if(target<=nums[0]){
            return 0;
        }
        else{
return right;}
    }
};